python_modules/dagster/dagster/_core/definitions/job_definition.py

```diff
@@ -828,6 +828,12 @@ def as_ephemeral_job(
             asset_selection=frozenset(asset_selection) if asset_selection else None,
         )
 
+    @property
+    def has_asset_selection(self) -> bool:
+        return bool(self.asset_selection) or (
+            self.asset_layer and bool(self.asset_layer.selected_asset_keys)
+        )
```

Comment:
i would call this "is_asset_job" given that's what it's actually trying to detect -- also don't think there's necessarily a reason to have the first part (`bool(self.asset_selection)`) here, other than i guess some theoretical mild perf improvements in certain cases

that being said, might be somewhat less relevant of a comment given my other comments